windows/mdag_enable_disable/enable_mdag.yml

```diff
@@ -0,0 +1,122 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Hardware requirements to run MDAG:
+# 1. CPU hardware virtualization is enabled.
+# 2. Minimum 4 cores (logical processors)
+# 3. Minimum 8-GB RAM
+#
+- name: "Initialize the VM hardware related variables"
+  ansible.builtin.set_fact:
+    vm_nested_virt_status: false
+    vm_initial_cpu_num: ""
+    vm_initial_mem_mb: ""
+
+- name: "Get CPU hardware virtualization status, CPU number and memory size"
+  include_tasks: ../../common/vm_get_config.yml
+  vars:
+    property_list: ['config.nestedHVEnabled', 'config.hardware.memoryMB', 'config.hardware.numCPU']
+
+- name: "Set facts of VM hardware related info"
+  ansible.builtin.set_fact:
+    vm_nested_virt_status: "{{ vm_config.config.nestedHVEnabled }}"
+    vm_initial_cpu_num: "{{ vm_config.config.hardware.numCPU }}"
+    vm_initial_mem_mb: "{{ vm_config.config.hardware.memoryMB }}"
+  when:
+    - vm_config.config is defined
+    - vm_config.config.nestedHVEnabled is defined
+    - vm_config.config.hardware.memoryMB is defined
+    - vm_config.config.hardware.numCPU is defined
+
+- name: "Edit VM Settings"
+  block:
+    - name: "Shutdown VM"
+      include_tasks: ../utils/win_shutdown_restart.yml
+      vars:
+        set_win_power_state: "shutdown"
+
+    - name: "Enable CPU hardware virtualization for the VM"
+      include_tasks: ../../common/vm_set_nested_virtual.yml
+      vars:
+        vm_nested_virt: true
+      when: not vm_nested_virt_status
+
+    - name: "Set VM CPU number to 4"
+      include_tasks: ../../common/vm_set_cpu_number.yml
+      vars:
+        num_cores_per_socket: 2
+        num_cpus: 4
+      when: vm_initial_cpu_num | int < 4
+
+    - name: "Set VM memory size to 8 GB"
+      include_tasks: ../../common/vm_set_memory_size.yml
+      vars:
+        memory_mb: 8192
+      when: vm_initial_mem_mb | int < 8 * 1024
+
+    - name: "Power on the VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'powered-on'
+
+    - name: "Update the inventory"
+      include_tasks: ../utils/win_update_inventory.yml
+  when: >
+    (not vm_nested_virt_status) or
+    (vm_initial_cpu_num | int < 4) or 
+    (vm_initial_mem_mb | int < 8 * 1024) 
```

Comment:
what if `vm_initial_cpu_num` or `vm_initial_mem_mb` is empty?

Reply:
Only when task vm_get_config.yml failed, they will be empty. Then the case will failed before the block.